pkg/git/fetch.go

```diff
@@ -47,9 +48,23 @@ func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
 		return errors.Wrap(err, "creating remote")
 	}
 
+	resolver := &remoteGitResolver{}
+	resolvedSourceConfig, err := resolver.Resolve(auth, corev1alpha1.SourceConfig{
+		Git: &corev1alpha1.Git{
+			URL:                  gitURL,
+			Revision:             gitRevision,
+			InitializeSubmodules: f.InitializeSubmodules,
+		},
+		SubPath: "",
+	})
+	if err != nil {
+		return errors.Wrap(err, "resolving source config")
+	}
+
 	err = remote.Fetch(&gogit.FetchOptions{
-		RefSpecs: []config.RefSpec{"refs/*:refs/*"},
+		RefSpecs: []config.RefSpec{config.RefSpec(resolvedSourceConfig.Git.Revision + ":" + resolvedSourceConfig.Git.Revision)},
```

Comment:
Configure RefSpecs to not pull EVERYTHING but instead just the specific resolved commit we want.